File: examples/manifest_runner/scripts/julia/sum.jl
using LinearAlgebra
using BenchmarkTools
using JSON3
include(joinpath(@__DIR__, "utils.jl"))

n = parse(Int, ARGS[1])

x = rand(n)
b = @benchmark sum($x) samples=100 evals=1

rst = trial_to_dict(b)
rst["size"] = n
rst["name"] = "sum"
rst["framework"] = "julia"

JSON3.write(stdout, rst)

File: examples/manifest_runner/scripts/python/sum.py
import random
import json
from utils import belapsed
import sys

n = int(sys.argv[1])
x = [random.random() for _ in range(n)]


def sum(x):
    rst = 0
    for x in x:
        rst += x
    return rst


rst = {
    "time": belapsed(lambda: sum(x), number=100),
    "name": "sum",
    "size": n,
    "framework": "python",
}

print(json.dumps(rst))
